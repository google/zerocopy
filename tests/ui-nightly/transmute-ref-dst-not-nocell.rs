// Copyright 2023 The Fuchsia Authors
//
// Licensed under a BSD-style license <LICENSE-BSD>, Apache License, Version 2.0
// <LICENSE-APACHE or https://www.apache.org/licenses/LICENSE-2.0>, or the MIT
// license <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your option.
// This file may not be copied, modified, or distributed except according to
// those terms.

include!("../../zerocopy-derive/tests/util.rs");

extern crate zerocopy;

use zerocopy::transmute_ref;

fn main() {}

#[derive(zerocopy::FromZeros, zerocopy::FromBytes)]
#[repr(transparent)]
struct Dst(AU16);

// `transmute_ref` requires that the destination type implements `NoCell`
const DST_NOT_NO_CELL: &Dst = transmute_ref!(&AU16(0));
