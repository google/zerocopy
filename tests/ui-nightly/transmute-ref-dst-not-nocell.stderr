error[E0277]: the trait bound `Dst: zerocopy::NoCell` is not satisfied
  --> tests/ui-nightly/transmute-ref-dst-not-nocell.rs:23:31
   |
23 | const DST_NOT_NO_CELL: &Dst = transmute_ref!(&AU16(0));
   |                               ^^^^^^^^^^^^^^^^^^^^^^^^
   |                               |
   |                               the trait `zerocopy::NoCell` is not implemented for `Dst`
   |                               required by a bound introduced by this call
   |
   = help: the following other types implement trait `zerocopy::NoCell`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and $N others
note: required by a bound in `AssertDstIsNoCell`
  --> tests/ui-nightly/transmute-ref-dst-not-nocell.rs:23:31
   |
23 | const DST_NOT_NO_CELL: &Dst = transmute_ref!(&AU16(0));
   |                               ^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `AssertDstIsNoCell`
   = note: this error originates in the macro `transmute_ref` (in Nightly builds, run with -Z macro-backtrace for more info)
